optuna/terminator/serror.py

```diff
@@ -0,0 +1,54 @@
+import abc
+
+import numpy as np
+
+from optuna.study.study import Study
+from optuna.trial._state import TrialState
+
+
+class BaseStatisticalErrorEvaluator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def evaluate(
+        self,
+        study: Study,
+    ) -> float:
+        pass
+
+
+class CrossValidationStatisticalErrorEvaluator(BaseStatisticalErrorEvaluator):
+    def __init__(self, user_attr_key: str = "cv_scores") -> None:
+        self._user_attr_key = user_attr_key
+
+    def evaluate(
+        self,
+        study: Study,
+    ) -> float:
+        if len(study.get_trials(states=(TrialState.COMPLETE,))) == 0:
+            raise ValueError(
+                "Statistical error cannot be calculated because no trial has been completed."
+            )
+
+        cv_scores = study.best_trial.user_attrs[self._user_attr_key]
+
+        k = len(cv_scores)
+        scale = 1 / k + 1 / (k - 1)
+
+        var = scale * np.var(cv_scores)
+
+        return float(var)
+
+
+class StaticStatisticalErrorEvaluator(BaseStatisticalErrorEvaluator):
+    def __init__(self, constant: float) -> None:
+        self._constant = constant
+
+    def evaluate(
+        self,
+        study: Study,
+    ) -> float:
+        if len(study.get_trials(states=(TrialState.COMPLETE,))) == 0:
+            raise ValueError(
+                "Statistical error cannot be calculated because no trial has been completed."
+            )
+
```

Comment:
I didn't understand why this error was needed, given the logic of simply returning `self._constant` but does it make sense in a broader context? A comment might be helpful for future maintainers.

Reply:
Thanks. I removed this check: https://github.com/optuna/optuna/pull/4405/commits/e93a62571272d4dfc32e3435aa0742d4e87abd61